### PR TITLE
GFORMS-2771 - Add insufficient credentials enrolment outcome...

### PR DESCRIPTION
**Pull request by jmartinhmrc**
...to block users with the assistant credential from completing the enrolment section.

### CODE REVIEW

File: app/uk/gov/hmrc/gform/gform/EnrolmentController.scala [hunk 1]
```diff
@@ -316,24 +368,30 @@ class EnrolmentController(
...
-              renderEnrolmentSection(
-                cache.formTemplate,
-                cache.retrievals,
-                enrolmentSection,
-                FormModelOptics.fromEnrolmentSection(enrolmentSection, cache),
-                Nil,
-                ValidationResult.empty
+      if (cache.retrievals.getCredentialRole.get == Assistant) {
+        Redirect(uk.gov.hmrc.gform.gform.routes.EnrolmentController.insufficientCredentialsPage(formTemplateId))
+          .flashing("formTitle" -> cache.formTemplate.formName.value)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/EnrolmentController.scala:373]: This param seems not to be used, let's remove it.

File: app/uk/gov/hmrc/gform/gform/EnrolmentController.scala [hunk 2]
```diff
@@ -270,6 +289,39 @@ class EnrolmentController(
...
+                    {
+                      case "change-gg-account" =>
+                        Redirect(
+                          uk.gov.hmrc.gform.gform.routes.NewFormController.dashboardWithNewSession(formTemplate._id)
+                        )
+                      case "sign-out" =>
+                        Redirect(uk.gov.hmrc.gform.gform.routes.SignOutController.signOut(formTemplate._id))
+                      case unknown =>
+                        throw new Exception(s"Unexpected value of linked account parameter: '$unknown'")
+                    }
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/EnrolmentController.scala:321]: This is now duplicated twice, can we extract it into a common method.

File: app/uk/gov/hmrc/gform/views/hardcoded/InsufficientCredentialsPage.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+package uk.gov.hmrc.gform.views.hardcoded
+
+import play.api.data.Form
+import play.api.i18n.Messages
+import play.twirl.api.Html
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.FormTemplate
+import uk.gov.hmrc.govukfrontend.views.html.components.{ ErrorLink, ErrorMessage, ErrorSummary, Fieldset, GovukErrorMessage, GovukFieldset, GovukHint, GovukLabel, GovukRadios, Legend, RadioItem, Radios, Text }
+
+class InsufficientCredentialsPage(val formTemplate: FormTemplate, form: Form[String])(implicit messages: Messages)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/views/hardcoded/InsufficientCredentialsPage.scala:25]: This looks like copy of `EnrolmentAlreadyLinkedPage`. Can we rename `EnrolmentAlreadyLinkedPage` to something more general and reuse it?

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/gform/EnrolmentController.scala (hunk 2 above)
Comment (jmartinhmrc) [reply app/uk/gov/hmrc/gform/gform/EnrolmentController.scala]: Done

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**